plugins/lua/zone.lua

```diff
@@ -1216,10 +1216,186 @@ function RetireLocationOverlay:confirm_retire()
         curry(retire, location))
 end
 
+-- -------------------
+-- AnimalActionsWidget
+--
+
+-- Make sure an animal unit of your civ is selected
+local function check_valid_unit()
+    local unit = dfhack.gui.getSelectedUnit(true)
+    return unit
+        and dfhack.units.isFortControlled(unit)
+        and dfhack.units.isAlive(unit)
+        and dfhack.units.isAnimal(unit)
+end
+
+local function is_geldable()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    return dfhack.units.isGeldable(unit)
+end
+
+local function is_not_pet()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    return not dfhack.units.isPet(unit)
+end
+
+local function is_avail_adoption()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    local creature_id = df.creature_raw.find(unit.race).creature_id
+    -- Make sure not already a pet and not a cat as they adopt owner
+    return not dfhack.units.isPet(unit) and creature_id ~= 'CAT'
+end
+
+local function is_tamable()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    return dfhack.units.isTamable(unit) and not dfhack.units.isDomesticated(unit)
+end
+
+AnimalActionsWidget=defclass(AnimalActionsWidget, overlay.OverlayWidget)
+AnimalActionsWidget.ATTRS {
+    desc = "Add options to tamed animals view sheet",
+    default_pos={x=-42,y=37},
+    default_enabled=true,
+    viewscreens='dwarfmode/ViewSheets/UNIT/Overview',
+    frame={w=23, h=6},
+}
+
+-- The above function already handles checking if valid unit
+-- so just set slaughter flag
+function AnimalActionsWidget:set_slaughter_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option then
+        unit.flags2.slaughter = option
+    else
+        unit.flags2.slaughter = option
+    end
```

Comment:
```suggestion
    unit.flags2.slaughter = option
```

Reply:
I now see what you were trying to say my bad